ddtrace/contrib/asgi/middleware.py

```diff
@@ -128,21 +116,31 @@ async def __call__(self, scope, receive, send):
         if sample_rate is not None:
             span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, sample_rate)
 
+        method = scope.get("method")
+        server = scope.get("server")
+        url = None
+        if server and len(server) == 2:
+            port = server[1]
+            server_host = server[0] + (":" + str(port) if port is not None and port != 80 else "")
+            full_path = scope.get("root_path", "") + scope.get("path", "")
+            url = scope.get("scheme", "http") + "://" + server_host + full_path
+
+        trace_utils.set_http_meta(self.integration_config, span, method=method, url=url)
+
         tags = _extract_tags_from_scope(scope, self.integration_config)
         span.set_tags(tags)
 
         store_request_headers(headers, span, self.integration_config)
 
         async def wrapped_send(message):
+            status_code = None
             if span and message.get("type") == "http.response.start":
                 if "status" in message:
                     status_code = message["status"]
```

Comment:
```suggestion
                status_code = message["status"]
            else:
                status_code = None
```

lol unforunately github doesn't support multiline suggestions so this got a little messy 😕